src/app/component/Slider/Slider.component.js

```diff
@@ -413,24 +447,43 @@ export class Slider extends PureComponent {
     }
 
     renderSliderContent() {
-        const { activeImage, children, isVertical } = this.props;
+        const {
+            children,
+            children: { length },
+            isVertical,
+            childrenClones
+        } = this.props;
+        const { activeImage } = this.state;
 
         if (!this.getIsSlider()) {
             return children;
         }
 
+        const slidesOnLeft = activeImage - this.getFocusSlide() + 1;
+        const firstLeftIndex = slidesOnLeft < 0 ? length + slidesOnLeft : slidesOnLeft;
+
+        const leftClone = firstLeftIndex === 0 ? length - 1 : firstLeftIndex - 1;
+        const rightClone = firstLeftIndex >= length ? 0 : firstLeftIndex;
+
+        const slides = this.getIsInfinity() ? [childrenClones[leftClone],
+            ...children.slice(firstLeftIndex, length),
+            ...children.slice(0, firstLeftIndex),
+            rightClone === leftClone ? null : childrenClones[rightClone]] : children;
```

Comment:
Unreadable, please refactor. Prefer using ternary only in very short cases, otherwise, you may use a function for easier branching declarations.